src/dune_rules/lib.ml

```diff
@@ -947,9 +949,15 @@ end = struct
           let open Resolve.Memo.O in
           let* vlib = resolve_forbid_ignore name in
           let virtual_ = Lib_info.virtual_ vlib.info in
-          match virtual_ with
-          | false -> Error.not_virtual_lib ~loc ~impl:info ~not_vlib:vlib.info
-          | true -> Resolve.Memo.return vlib
+          let is_parameter = Lib_info.is_parameter vlib.info in
```

Comment:
Since being a virtual library and a parameter is mutually exclusive, what about introducing a type to reflect that and returning it from some sort of a `Lib_info.kind` function. The sooner this invariant is enforced, the easier it will be to make sure we never make this mistake.

Reply:
Do you mean that we should duplicate the virtual type with a parameter equivalent or do you mean we should have a `Parameter` kind? 

If the former, `Lib_info.is_parameter is just an alias for `t.kind = parameter`.

Reply:
I argue that we need a function as follows:

```
val lib_kind : [`Normal | `Virtual | `Parameter | `Ppx_deriver ..]
```

The function `Lib_info.is_parameter` shouldn't be necessary at all.

Now you might say that we've done a similar sort of mistake with `Lib_info.virtual_`. If we didn't have this function, I think it would be simpler for us to be confident that we're correctly making this change everywhere.

Reply:
Would "the right way" to do to this be to extend https://github.com/maiste/dune/blob/a45b410b28dd1d8de56f3a5a0f5bdb0a460aae5a/src/dune_lang/lib_kind.mli#L16-L20 to include a variant for `Virtual`? Or is virtual not disjoint with the alternatives?

Reply:
`Virtual` should be disjoint from the rest.